go/teams/list.go

```diff
@@ -45,6 +45,113 @@ func getTeamsListFromServer(ctx context.Context, g *libkb.GlobalContext, uid key
+}
+
+// verifyMemberRoleInTeam checks if role give in MemberInfo matches
+// what team chain says. Nothing is checked when MemberInfo's role is
+// NONE, in this context it means that user has implied membership in
+// the team and no role given in sigchain.
+func verifyMemberRoleInTeam(ctx context.Context, member keybase1.MemberInfo, team *Team) (expected bool, err error) {
+	if member.Role == keybase1.TeamRole_NONE {
+		return true, nil
+	}
+
+	memberUV, err := team.chain().GetLatestUVWithUID(member.UserID)
+	if err != nil {
+		return false, err
+	}
+	role, err := team.chain().GetUserRole(memberUV)
+	if err != nil {
+		return false, err
+	}
+	if role != member.Role {
+		return false, fmt.Errorf("unexpected member role: got %v but actual role is %v", member.Role, role)
+	}
+	return true, nil
+}
+
+// getTeamForMember tries to load team in a recent enough state to
+// contain member with correct role as set in MemberInfo. It might
+// trigger a reload with ForceRepoll if cached state does not match.
+func getTeamForMember(ctx context.Context, g *libkb.GlobalContext, member keybase1.MemberInfo, needAdmin bool) (*Team, error) {
+	team, err := Load(ctx, g, keybase1.LoadTeamArg{
+		ID:          member.TeamID,
+		NeedAdmin:   needAdmin,
+		ForceRepoll: false,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	expectedRole, _ := verifyMemberRoleInTeam(ctx, member, team)
+	if !expectedRole {
+		team, err = Load(ctx, g, keybase1.LoadTeamArg{
+			ID:          member.TeamID,
+			NeedAdmin:   needAdmin,
+			ForceRepoll: true,
+		})
+		if err != nil {
+			return nil, err
+		}
+
+		_, err := verifyMemberRoleInTeam(ctx, member, team)
+		if err != nil {
+			return nil, fmt.Errorf("server was wrong about role in team : %v", err)
+		}
+	}
+
+	return team, nil
+}
+
+func getUserAndFullName(ctx context.Context, g *libkb.GlobalContext, uid keybase1.UID) (username libkb.NormalizedUsername, fullName string, err error) {
+	username, err = g.GetUPAKLoader().LookupUsername(ctx, uid)
+	if err != nil {
+		return "", "", err
+	}
+	fullName, err = engine.GetFullName(ctx, g, uid)
+	if err != nil {
+		return "", "", err
+	}
+
+	return username, fullName, err
+}
+
+func fillUsernames(ctx context.Context, g *libkb.GlobalContext, res *keybase1.AnnotatedTeamList) error {
+	var userList []keybase1.UID
+	userSet := map[keybase1.UID]int{}
+
+	for _, member := range res.Teams {
+		_, ok := userSet[member.UserID]
+		if ok {
+			continue
+		}
+
+		userSet[member.UserID] = len(userList)
+		userList = append(userList, member.UserID)
+	}
+
+	namePkgs, err := g.UIDMapper.MapUIDsToUsernamePackages(ctx, g, userList, 0, 0, true)
+	if err != nil {
+		return err
+	}
+
+	for id := range res.Teams {
+		member := &res.Teams[id]
+		num := userSet[member.UserID]
+		pkg := namePkgs[num]
+
+		member.Username = pkg.NormalizedUsername.String()
+		if fullName := pkg.FullName; fullName != nil {
+			member.FullName = string(fullName.FullName)
+		}
```

Comment:
can this just be

    if pkg.FullName != nil { member.FullName = string(pkg.FullName) }

?  or am I missing something?

Reply:
`pkg.FullName` is a pointer to `FullNamePackage` that has `FullName` field of type string (or rather `type FullName string`)

I'm changing this to:
```
member.Username = pkg.NormalizedUsername.String()
if pkg.FullName != nil {
	member.FullName = string(pkg.FullName.FullName)
}
```
which seems to work